File: src/main/java/com/matamercer/microblog/utilities/EnvironmentUtil.java
package com.matamercer.microblog.utilities;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Component;

import java.net.InetAddress;
import java.net.UnknownHostException;

@Component
public class EnvironmentUtil {
    @Autowired
    Environment environment;

    private String port;
    private String hostname;

    public String getPort() {
        if (port == null) port = environment.getProperty("local.server.port");
        return port;
    }

    public Integer getPortAsInt() {
        return Integer.valueOf(getPort());
    }

    public String getHostname() throws UnknownHostException {
        if (hostname == null) hostname = InetAddress.getLocalHost().getHostAddress();
        return hostname;
    }

    public String getServerUrl() throws UnknownHostException {
        return "https://" + getHostname() + ":" + getPort();
    }

}

File: src/main/java/com/matamercer/microblog/utilities/EnvironmentUtil.kt
package com.matamercer.microblog.utilities

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.env.Environment
import org.springframework.stereotype.Component
import java.net.InetAddress
import java.net.UnknownHostException

@Component
class EnvironmentUtil {
    @Autowired
    var environment: Environment? = null
    private var port: String? = null
        get() {
            if (field == null) field = environment!!.getProperty("local.server.port")
            return field
        }

    @get:Throws(UnknownHostException::class)
    private var hostname: String? = null
        get() {
            if (field == null) field = InetAddress.getLocalHost().hostAddress
            return field
        }
    private val portAsInt: Int
        get() = Integer.valueOf(port)

    @get:Throws(UnknownHostException::class)
    val serverUrl: String
        get() = "http://$hostname:$port"
}